tests/net/lib/dns_cache/prj.conf

```diff
@@ -0,0 +1,15 @@
+CONFIG_ZTEST=y
+
+CONFIG_NETWORKING=y
+CONFIG_NET_TEST=y
+CONFIG_NET_L2_ETHERNET=n
+
+CONFIG_MAIN_STACK_SIZE=1344
+CONFIG_DNS_RESOLVER=y
+CONFIG_DNS_RESOLVER_CACHE=y
+CONFIG_LOG=y
+CONFIG_NET_LOG=y
+CONFIG_DNS_RESOLVER_LOG_LEVEL_DBG=y
+
+CONFIG_UBSAN=y
+CONFIG_ASAN=y
```

Comment:
IMO default `prj.conf` shoud be generic, POSIX-specifc configs should either go in a custom config file, or even better, as  `extra_configs` for posix target in `testcase.yml`.